torchqdynamics/odeint.py

```diff
@@ -1,80 +1,124 @@
 import warnings
+from abc import ABC, abstractmethod
+from enum import Enum
+from typing import List
 
 import torch
 
 from .solver import AdaptativeStep, FixedStep
 
 
-def odeint(qsolver, y0, tsave, sensitivity='autograd', parameters=None):
-    # check arguments
-    tsave = torch.tensor(tsave)
-    check_tsave(tsave)
-    if (parameters is not None) and (sensitivity in [None, 'autograd']):
-        warnings.warn('Argument `parameters` was supplied in `odeint` but not used.')
+class ForwardQSolver(ABC):
+    @abstractmethod
+    def forward(self, t, dt, rho):
+        pass
 
-    # dispatch to appropriate odeint subroutine
-    if sensitivity is None:
-        return odeint_inplace(qsolver, y0, tsave)
-    elif sensitivity == 'autograd':
-        return odeint_main(qsolver, y0, tsave)
-    elif sensitivity == 'adjoint':
-        return odeint_adjoint(qsolver, y0, tsave, parameters)
 
+class GradientAlgorithm(Enum):
+    AUTOGRAD = 'autograd'  # gradient computed by torch
+    NONE = 'none'  # don't compute the gradients
+    ADJOINT = 'adjoint'  # compute the gradient using the adjoint method
+
+
+def odeint(
+    qsolver,
+    y0,
+    t_save: torch.Tensor,
```

Comment:
I prefer this because it's consistent with the general "rule" of using snake case for variable names, and IMHO new exceptions add to the cognitive load. But if I'm the only one we can change CC @gautierronan @eliegenois

Reply:
I prefer `tsave`, but if there's a general rule that should be followed then I'm fine with `t_save` as well.